### ISSUE TITLE
[Avatar group][A11Y]: [A] 2.5.3 The accessible name of any interactive element should contain its visible text label.

### ISSUE THREAD

**Issue by MaksymZahreba** [UI5/webcomponents#11564]
### Bug Description

Issue found by a11y tool Siteimprove, caused by difference in naming of button and its label.

Note: in v1 button in shadow dom did not have the label attribute and issues is not triggered

### Affected Component

Avatar Group

### Expected Behaviour

No [A] level a11y issues found.

### Steps to Reproduce

1. launch Siteimprove tool on page with avatar group component

### Log Output, Stack Trace or Screenshots

Individual type set:
![Image](https://github.com/user-attachments/assets/2e02da69-024c-461c-b830-a6f62ec7289b)

![Image](https://github.com/user-attachments/assets/c6b7d832-90dc-4cd3-8b11-fc5c33294a1d)

Same issue is triggered for Group type:

![Image](https://github.com/user-attachments/assets/6d697a02-bf6b-4fb2-87e0-1518a0aea412)

![Image](https://github.com/user-attachments/assets/e1368c98-ed5a-4fef-91ec-05d1d62ff86b)

### Priority

Medium

### UI5 Web Components Version

2.10.0

### Browser

Chrome

### Organization

SAP \ UDEx

### Declaration

- [x] I’m not disclosing any internal or sensitive information.

**Comment by dobrinyonkov** [comment]
Hi @MaksymZahreba,

We usually check accessibility using Axe and AMP (Access Assistant). I just ran both on the Avatar Group component and neither showed any issues with the labels.

Can you point us to a specific WCAG requirement that isn’t met? It would also help if you explain how this affects users. That way, we can understand the impact and prioritize the fix.

Also, just to confirm: are you expecting the "+N" button (that shows how many avatars are hidden) to include the "+N" text in the label - for both individual and group types?

Thanks.
Dobrin

**Comment by MaksymZahreba** [comment]
Hi @dobrinyonkov ,
The WCAG requirement 2.5.3
https://www.w3.org/WAI/WCAG22/Understanding/label-in-name.html

This for example affects users that use voice controls which makes them unable to interact with elements due to difference in label and displayed content.

The set text to display and its `aria-label` should be the same. To add description for elements it is recommended to use `<aria-labeledby>` 

//rough code example//
`<ui5-button class="ui5-avatar-group-overflow-btn ui5-avatar-group-overflow-btn-s" ui5-button="" design="Default" type="Button" accessible-role="Button" desktop="" id="1" aria-labelledby="1 2">+6</ui5-button>` - here added id="1" and aria-labelledby="1 2" , removed accessible-name
`<button type="button" class="ui5-button-root" data-sap-focus-ref="true" tabindex="0" part="button" role="button">`  - aria-label removed
`<span id="2">more avatars</span>` - added, it is supposed to be hidden with CSS styles and used for aria-labeledby

![Image](https://github.com/user-attachments/assets/29ba5290-68e6-40a9-88bc-beb4f66b9376)

![Image](https://github.com/user-attachments/assets/e63578f3-c240-4e74-96fb-8007ba0a3b3f)

Regards,
Maksym

**Comment by MaksymZahreba** [comment]
Hi, any progress on the issue?

**Comment by dobrinyonkov** [comment]
Hello @MaksymZahreba,

Apologies for the late response. In the latest version of the Avatar Group:

- For type `individual`, the overflow button is labeled:
“+N Activate for complete list.”

<img width="510" height="38" alt="Image" src="https://github.com/user-attachments/assets/8a994a18-e133-4800-b17d-9293b70cd155" />

- For type `group`, there was no change as the whole group was labeled with custom text:
“Conjoined avatars. **Y** displayed, **N** hidden. Activate for complete list.”
This ensures users are clearly informed about how many avatars are shown and hidden.

I hope this helps. Let me know of any additional concerns. 

Kind regards,
Dobrin

**Comment by PlutaKatarzyna** [comment]
Hi @dobrinyonkov , while aria label is present to describe existing avatars and N+ avatar presence in the group, it doesn't match the text on the avatars itself, which is "visible" for a user and result with error with automation accessibility testing tools like SiteImprove.
While using screen reader (for example voiceOver) and go to individual avatars in a group (for me it goes with D key), it is read twice - one on inner text and second on avatar. I believe that one of it should have aria-hidden and it would probably resolve SiteImprove issue when this text doesn't match your aria description.
Please reopen the issue and consider solution.

https://github.com/user-attachments/assets/770fe527-ae35-41aa-8902-4be87db6bd07

**Comment by dobrinyonkov** [comment]
Hi @PlutaKatarzyna, this seems to be a different issue than the one described above. Could you please open a new issue for it so it can trackeed properly?

Thanks!

**Comment by dobrinyonkov** [comment]
Sorry I think I just now understood the issue, I have a question though: 

Do you suggest to make all of the avatars hidden (using aria-hidden) or some other element inside the avatar group?

Thanks!

**Comment by PlutaKatarzyna** [comment]
@dobrinyonkov I am not 100% sure, but when you see the recording, you can see that single Avatar (in avatars group) reads `.ui5-avatar-root` aria-label and inner text (.ui5-avatar-initials). This is just my guess, that `ui5-avatar-initials` should have aria-hidden to make sure only label will be read and intractable for screen reader.
<img width="811" height="110" alt="Image" src="https://github.com/user-attachments/assets/ca6124f3-b225-46c7-ace1-40778bfe066c" />
I am saying this is a guess, because it is not reproducible for single Avatar component. Only for Avatar group.

**Comment by PlutaKatarzyna** [comment]
Hi @dobrinyonkov is there a chance to reopen the issue?
